lib/screens/v2/widget_instance/reconstructed_alert.ex

```diff
@@ -177,7 +180,36 @@ defmodule Screens.V2.WidgetInstance.ReconstructedAlert do
   end
 
   # Given an entity and the directionality of the alert from the home stop,
-  # return a tuple with the affected direction_id and route id
+  # return a tuple with the affected direction_id and route_id
+
+  # If the route is red and the alert is downstream, we have to figure out whether the alert
+  # only affects one branch or both
+  defp get_direction_and_route_from_entity(
+         %{direction_id: nil, route: "Red", stop: stop_id},
+         location
+       )
+       when location in [:downstream, :boundary_downstream] do
+    cond do
+      Stop.on_ashmont_branch?(stop_id) and location in [:downstream, :boundary_downstream] ->
+        {0, "Red-Ashmont"}
+
+      Stop.on_ashmont_branch?(stop_id) ->
+        {1, "Red-Ashmont"}
+
+      Stop.on_braintree_branch?(stop_id) and location in [:downstream, :boundary_downstream] ->
+        {0, "Red-Braintree"}
+
+      Stop.on_braintree_branch?(stop_id) ->
+        {1, "Red-Ashmont"}
```

Comment:
Did you mean for this to be
```suggestion
        {1, "Red-Braintree"}
```
?